x/sequencer/ante/tx_arweave_block_verify.go

```diff
@@ -0,0 +1,60 @@
+package ante
+
+import (
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/warp-contracts/sequencer/x/sequencer/types"
+)
+
+// It checks if the transaction representing an Arweave block contains messages in the following order:
+// - the first message is block information (MsgArweaveBlockInfo)
+// - subsequent messages are L1 interactions (MsgArweaveTransaction)
+type ArweaveBlockTxDecorator struct {
+}
+
+func NewArweaveBlockTxDecorator() ArweaveBlockTxDecorator {
+	return ArweaveBlockTxDecorator{}
+}
+
+func (abtd ArweaveBlockTxDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	return ctx, verifyArweaveBlockTx(tx)
+}
+
+func verifyArweaveBlockTx(tx sdk.Tx) error {
+	msgs := tx.GetMsgs()
+	containsBlockInfoMsg := false
+
+	for i, msg := range msgs {
```

Comment:
go has also an index notation, msgs[0] will work, so it's possible to simplify this code